### DISCUSSION TITLE
Possibility to change Model in request?

### DISCUSSION THREAD

**Discussion by jochemgruter** [livewire/livewire#2879] [1. Help]
For security reasons, is it possible for the enduser to change the model in the livewire request?

For example, if I mount my component like this:
```
public $order;

public function mount(Order $order){
    $this->order = $order;
}
```

Can I then be sure that this order instance won't changed if the user tries to manipulate the request?

**Comment by gwleuverink** [comment]
Livewire stores a checksum of the current public state of your component. If it's tampered with in any way this mechanism should throw a CorruptComponentPayloadException.

https://github.com/livewire/livewire/blob/master/src/HydrationMiddleware/SecureHydrationWithChecksum.php
https://github.com/livewire/livewire/blob/master/tests/Unit/ComponentsAreSecureTest.php
